src/components/ArticlePage.jsx

```diff
@@ -1,31 +1,54 @@
 import { useParams } from "react-router-dom";
-import { getArticleById } from "../api";
+import { getArticles } from "../api";
 import { useState, useEffect } from "react";
+import { CommentCards } from "./CommentCards";
+import { Grid2, Container } from "@mui/material";
 import "./articlepage.css";
+import { getUsers } from "../api";
+
 export const ArticlePage = () => {
   const { articleId } = useParams();
   const [article, setArticle] = useState(null);
   const [loading, setLoading] = useState(true);
+  const [comments, setComments] = useState([]);
+  const [users, setUsers] = useState([]);
 
   useEffect(() => {
-    getArticleById(articleId)
+    getArticles(articleId)
       .then(({ article }) => {
-        console.log(article);
         setArticle(article);
         setLoading(false);
       })
       .catch((error) => {
-        console.error("Error fetching article:", error);
         setLoading(false);
+        return <p className="loading-info">{error}</p>;
       });
   }, [articleId]);
 
+  useEffect(() => {
+    getArticles(articleId, "getComments")
+      .then(({ comments }) => {
+        setComments(comments);
+        setLoading(false);
+      })
+      .catch((error) => {
+        setLoading(false);
+        return <p className="loading-info">{error}</p>;
```

Comment:
we really shouldn't return in a useEffect this will cause un expected errors.
If you want to display an error to users, think about using a state to know when there has been an error.